exec/cnex/nstring.c

```diff
@@ -195,6 +195,19 @@ BOOL string_startsWith(TString *self, TString *s)
     return TRUE;
 }
 
+int32_t string_findChar(TString *self, char c)
+{
+    size_t i;
+
+    for (i = 0; i < self->length; i++) {
+        if (self->data[i] == c) {
+            return (int32_t)i;
```

Comment:
You know how I feel about random casts like this. Why not make this function return a `size_t`?

Reply:
I wanted to do this on purpose so that I can return a -1 if something isn't found.  Is there a better way to handle this?  Since size_t is unsigned, I wanted to avoid making it return a size_t.

Reply:
There's a couple of ways you can do this while returning the index in a `size_t`:

- Return `SIZE_MAX` if something isn't found.
- Return a "found" flag in addition to the position.

There aren't very many elegant ways to handle this in C.

Reply:
I agree, fully; which is why I thought this method would be ok.  However, I suppose it could return a SIGNED integer, and then just return SIZE_MAX; since no string is ever going to have the dot out at SIZE_MAN that I should ever have to worry about!